All data and MC GTs updated with GEM electronics map

arunhep opened the pull request:
This PR updates the data and mc global tags with the GEM electronics map.
offline data GTs now also contain latest 2017 rereco conditions.

# Summary of changes in Global Tags 
 
## RunI data 
 
   * **RunI HLT processing** : [100X_dataRun2_HLT_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v2) as [100X_dataRun2_HLT_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_frozen_v2/100X_dataRun2_HLT_frozen_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunI Offline processing** : [100X_dataRun2_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v3) as [100X_dataRun2_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_v3/100X_dataRun2_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

## RunII data 
 
   * **RunII HLTHI processing** : [100X_dataRun2_HLTHI_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLTHI_frozen_v2) as [100X_dataRun2_HLTHI_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLTHI_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLTHI_frozen_v2/100X_dataRun2_HLTHI_frozen_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunII HLT relval processing** : [100X_dataRun2_HLT_relval_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_relval_v2) as [100X_dataRun2_HLT_relval_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_relval_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_relval_v2/100X_dataRun2_HLT_relval_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunII Offline processing** : [100X_dataRun2_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v3) as [100X_dataRun2_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_v3/100X_dataRun2_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

   * **RunII Offline relval processing** : [100X_dataRun2_relval_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_relval_v3) as [100X_dataRun2_relval_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_relval_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_relval_v3/100X_dataRun2_relval_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

   * **RunII Prompt processing** : [100X_dataRun2_PromptLike_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_PromptLike_v2) as [100X_dataRun2_PromptLike_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_PromptLike_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_PromptLike_v2/100X_dataRun2_PromptLike_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunI HLT processing** : [100X_dataRun2_HLT_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v2) as [100X_dataRun2_HLT_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_frozen_v2/100X_dataRun2_HLT_frozen_v1): 
      * update of GEM electronics map
      * change of SiPixel templates 0T tag

## Upgrade 
 
   * **PhaseII realistic scenario** : [100X_upgrade2023_realistic_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2023_realistic_v2) as [100X_upgrade2023_realistic_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2023_realistic_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2023_realistic_v2/100X_upgrade2023_realistic_v1): 
      * update of GEM electronics map

   * **PhaseI 2018 cosmics scenario** : [100X_upgrade2018cosmics_realistic_deco_v10](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v10) as [100X_upgrade2018cosmics_realistic_deco_v9](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v9) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v10/100X_upgrade2018cosmics_realistic_deco_v9): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PhaseI 2018 design scenario** : [100X_upgrade2018_design_IdealBS_v8](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_design_IdealBS_v8) as [100X_upgrade2018_design_IdealBS_v7](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_design_IdealBS_v7) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018_design_IdealBS_v8/100X_upgrade2018_design_IdealBS_v7): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PhaseI 2018 realistic scenario** : [100X_upgrade2018_realistic_v12](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_realistic_v12) as [100X_upgrade2018_realistic_v11](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_realistic_v11) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018_realistic_v12/100X_upgrade2018_realistic_v11): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PostLS2 design scenario** : [100X_postLS2_design_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_design_v3) as [100X_postLS2_design_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_design_v2) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_design_v3/100X_postLS2_design_v2): 
      * update of Jet probability b tagging calibrations
      * update of GEM electronics map

   * **PostLS2 realistic scenario** : [100X_postLS2_realistic_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_realistic_v3) as [100X_postLS2_realistic_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_realistic_v2) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_realistic_v3/100X_postLS2_realistic_v2): 
      * update of Jet probability b tagging calibrations
      * update of GEM electronics map

## 2017_MC 
 
   * **PhaseI 2017 cosmics peak scenario** : [100X_mc2017cosmics_realistic_peak_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_peak_v4) as [100X_mc2017cosmics_realistic_peak_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_peak_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017cosmics_realistic_peak_v4/100X_mc2017cosmics_realistic_peak_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 cosmics scenario** : [100X_mc2017cosmics_realistic_deco_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_deco_v4) as [100X_mc2017cosmics_realistic_deco_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_deco_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017cosmics_realistic_deco_v4/100X_mc2017cosmics_realistic_deco_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 design scenario** : [100X_mc2017_design_IdealBS_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_design_IdealBS_v4) as [100X_mc2017_design_IdealBS_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_design_IdealBS_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017_design_IdealBS_v4/100X_mc2017_design_IdealBS_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 realistic scenario** : [100X_mc2017_realistic_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_realistic_v4) as [100X_mc2017_realistic_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_realistic_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017_realistic_v4/100X_mc2017_realistic_v3): 
      * update of GEM electronics map

arunhep committed:
data GTs updated with GEM electronics map

cmsbuild commented:
A new Pull Request was created by  @arunhep (Arun Kumar) for CMSSW_10_0_X.

It involves the following packages:

Configuration/AlCa


@ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @lpernie can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @mmusich, @ghellwig, @tocheng this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

arunhep commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ef6129341a3df7d2829f22ca90ce9c8ed5c47a65
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26810/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26810/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2468519
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2468349
 - DQMHistoTests: Total skipped: 169
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.980000000156 KiB( 22 files compared)
 - Checked 111 log files, 9 edm output root files, 27 DQM output files

arunhep commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_0_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_10_1_X is complete. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

arunhep committed:
MC GT updates with GEM electronics map

arunhep committed:
switch the reading of GEM electronics map from DB

cmsbuild commented:
Pull request #22592 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @slava77, @lpernie can you please check and sign again.

arunhep commented:
as agreed in ORP today, we have updated all relevant MC GTs with the same GEM electronics map as data tags and switched the flag to True to enable the reading of electronics map from DB

arunhep commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f1ff1f1639360418fca816cc7a97d030ecc0f68c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26834/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26834/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 26 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2468519
 - DQMHistoTests: Total failures: 94
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2468256
 - DQMHistoTests: Total skipped: 169
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.25000000001 KiB( 22 files compared)
 - Checked 111 log files, 9 edm output root files, 27 DQM output files

arunhep commented:
+1

jshlee commented:
thank you @arunhep
As far as I was aware, the current GT will only work for slice tests.
I did ask for a full geometry MC mapping DB, but I dont think this is it.

For now, I think it would be better to just change useDBEMap to true using Modifier_run2_GEM_2017_cff.

arunhep commented:
@jshlee 

> As far as I was aware, the current GT will only work for slice tests.
> I did ask for a full geometry MC mapping DB, but I dont think this is it.
> 
> For now, I think it would be better to just change useDBEMap to true using Modifier_run2_GEM_2017_cff.

but how will that be different from changing the flag in gemUnpacker_cfi.py ?

perrotta commented:
Using the modifier for run2_GEM_2017 allows it being set only for Run2_2017 and Riun2_2018, while it remains "false" for all other periods (namely Run3 and Phase2). As of the current implementation here the usage of DB mapping is also allowed in the gem unpacker for Run3 and Phase 2 workflows. Therefore, if the mapping is not present, or wrong, for whatever else that Run2, the flag change should be restricted by that modifier

@jshlee @arunhep  , since it does not make crashing the Run3 and Phase2 workflows in the matrix, I suppose that something is present in the GT also for those eras, although possibly the wrong one: could you please explain what should we expect there?

Indeed, looking at the jenkins outputs of this PR, we can see that impact parameters get changed in the 2019 wf (11624.0):
![image](https://user-images.githubusercontent.com/4069749/37393172-a4892198-2770-11e8-99e3-91c7f7786cff.png)
(and BTag as well, of course). It looks like being originated by the switch, as it did not show up in the first tests of this PR when getting the map from DB was not allowed yet. 

@jshlee : 
- How can the GEM enter in the computation of that IP? (Let me remind that in Run3 and Phase2 eras the GEM local reco is allowed)
- Why we do not see the same also in the Phase2 wfs?

arunhep commented:
@perrotta 

> Indeed, looking at the jenkins outputs of this PR, we can see that impact parameters get changed in the 2019 wf (11624.0):

its not because of GEM but also because for this there is also an update of jet probability b tagging calibrations : https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_realistic_v3/100X_postLS2_realistic_v2

but for phase-2 only update is GEM electronics map : 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2023_realistic_v2/100X_upgrade2023_realistic_v1

i should add the description of GT changes..

perrotta commented:
@arunhep : thank you, that makes sense. In fact, I did not notice that you added another commit yesterday besides the one that sets ```useDBEMap = cms.bool(True)```

perrotta commented:
Still, @jshlee please comment about the usage of the current DB map in Run3 and Phase2 wfs

jshlee commented:
The current DB would only pack and unpack gem hits for the slice test chambers only.
For this PR, it would be best to use Modifier_run2_GEM_2017_cff to change the useDBEMap to true.

I will test this further and report back

jshlee commented:
just realised that gemUnpacker_cfi.py is not needed or used.

process.muonGEMDigis is the one that needs to be changed in
https://github.com/cms-sw/cmssw/blob/master/Configuration/StandardSequences/python/RawToDigi_cff.py#L113

slava77 commented on EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py:
```diff
@@ -3,5 +3,5 @@
     InputLabel = cms.InputTag("rawDataCollector"),
     UnpackStatusDigis = cms.bool(False),
-    useDBEMap = cms.bool(False),
+    useDBEMap = cms.bool(True),
```
is this change in the master already?

jshlee replied:
no. 
and this file isnt used and should be removed.

jshlee commented:
@arunhep - i will fix the gem issue with backport of #22619
can you remove the changes applied to EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py

arunhep committed:
revert back the gem unpacker from DB flag

cmsbuild commented:
Pull request #22592 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @lpernie can you please check and sign again.

arunhep commented:
@jshlee 
> can you remove the changes applied to EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py

done

perrotta commented:
... and as such reco signature is not needed any more
The reco part of the adaptation of the GEMs to Run2 is now addressed by #22619

arunhep commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

arunhep commented:
tests are triggered just for the sake of confirmation.

cmsbuild commented:
+1
Tested at: 1e4c30fbfc11cca44cc05a334f931bb9c0d69f66
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26868/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/26868/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 24 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2468519
 - DQMHistoTests: Total failures: 94
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2468256
 - DQMHistoTests: Total skipped: 169
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.48999999997 KiB( 22 files compared)
 - Checked 111 log files, 9 edm output root files, 27 DQM output files

arunhep commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_0_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_10_1_X is complete. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

davidlange6 commented:
Your PR is unmergeable. Please have a look and possibly rebase it.

arunhep committed:
Merge branch 'CMSSW_10_0_X' into 100X_GEMElectronicsMap

cmsbuild commented:
Pull request #22592 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @lpernie can you please check and sign again.

arunhep commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2a8ce2f94af6dca72a2a13649721d6c8896121a8
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/27121/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2a8ce2f94af6dca72a2a13649721d6c8896121a8
https://github.com/cms-sw/cmssw/commit/6b5f29c5052e43577e0f2177a66dd4716f80b97c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/27121/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/27121/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22592/27121/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 28 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2468935
 - DQMHistoTests: Total failures: 94
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2468672
 - DQMHistoTests: Total skipped: 169
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.980000000083 KiB( 22 files compared)
 - Checked 111 log files, 9 edm output root files, 27 DQM output files

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_0_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_10_1_X is complete. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

arunhep commented:
@fabiocos what are the plans for this PR? I was hoping it to be integrated in 10.0.5
thanks.

fabiocos commented:
@arunhep it looked lated for 10_0_5, whose main target was different. I will merge it, although I hope that operations move soon after Easter to 10_1_X

fabiocos commented:
+1
